cmd/dev.go

```diff
@@ -224,14 +250,48 @@ var devCmd = &cobra.Command{
 	},
 }
 
+func checkHostnames() error {
+	hostnames := []string{compose.HostLocalDashboardNhostRun, compose.HostLocalGraphqlNhostRun, compose.HostLocalAuthNhostRun, compose.HostLocalStorageNhostRun, compose.HostLocalFunctionsNhostRun, compose.HostLocalMailNhostRun}
+
+	for _, hostname := range hostnames {
+		_, err := net.LookupIP(compose.HostLocalDashboardNhostRun)
+		if err != nil {
+			logger.DEBUG = false
+			fmt.Println(fmt.Sprintf(`Failed to resolve '%s' hostname
+
+Make sure that you have internet connection.
+
+If you want to use CLI offline, make sure to add the following lines in /etc/hosts:
+
+%s
+`, hostname, offlineConfigForSslHostnames(hostnames)))
+			return fmt.Errorf("failed to resolve '%s' hostname: %v", hostname, err)
+		}
+	}
+
+	return nil
+}
+
+func offlineConfigForSslHostnames(hostnames []string) string {
+	lines := []string{}
+	for _, hostname := range hostnames {
+		lines = append(lines, fmt.Sprintf("127.0.0.1	%s", hostname))
+	}
+	return strings.Join(lines, "\n")
+}
+
 func getPorts(fs *flag.FlagSet) (*ports.Ports, error) {
-	var proxyPort, dbPort, graphqlPort, hasuraConsolePort, hasuraConsoleApiPort, smtpPort, minioS3Port, mailhogPort, dashboardPort uint32
+	var proxyPort, sslProxyPort, dbPort, graphqlPort, hasuraConsolePort, hasuraConsoleApiPort, smtpPort, minioS3Port uint32
```

Comment:
I suggest using a declaration block for better readability and renaming `hasuraConsoleApiPort` to `hasuraApiPort`.